superset-frontend/src/components/Icons/BaseIcon.tsx

```diff
@@ -46,14 +46,12 @@ export const BaseIconComponent: React.FC<
   ...rest
 }) => {
   const theme = useTheme();
-  const iconCss = css`
-    color: ${iconColor || theme.colorIcon};
```

Comment:
This was preventing some magic to happen in the Button component that aligns icons with the button style